libr/bin/format/elf/elf.c

```diff
@@ -2371,7 +2371,10 @@ struct Elf_(r_bin_elf_obj_t)* Elf_(r_bin_elf_new_buf)(RBuffer *buf) {
 	struct Elf_(r_bin_elf_obj_t) *bin = R_NEW0 (struct Elf_(r_bin_elf_obj_t));
 	bin->kv = sdb_new0 ();
 	bin->b = r_buf_new ();
-	bin->size = buf->length;
+//	bin->size = buf->length;
+	ut64 curr = lseek (buf->fd, 0, SEEK_CUR);
```

Comment:
it is a bad idea to use buf->fd, or even lseek directly. those vars are suposed to be private to the abstraction of RBuffer.  an rbuffer can even work without an fd, so it may segfault or produce wrong results.